strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/layout/NugetHierarchicalLayoutProvider.java

```diff
@@ -60,40 +52,15 @@ public NugetHierarchicalArtifactCoordinates getArtifactCoordinates(String path)
     }
 
     @Override
-    public ArtifactInputStream getInputStream(String storageId,
-                                              String repositoryId,
-                                              String path)
-        throws IOException,
-        NoSuchAlgorithmException,
-        ArtifactTransportException
+    protected boolean isMetadata(String path)
     {
-        Storage storage = getConfiguration().getStorage(storageId);
-
-        logger.debug("Checking in " + storage.getId() + ":" + repositoryId + "...");
-
-        Repository repository = storage.getRepository(repositoryId);
-        StorageProvider storageProvider = storageProviderRegistry.getProvider(repository.getImplementation());
-
-        String storagePath = storage.getRepository(repositoryId).getBasedir() + '/' + path;
-        ArtifactInputStream ais = storageProvider.getInputStreamImplementation(storagePath);
-        
-        return ais;
+        return path.endsWith("nuspec");
     }
-
+    
     @Override
-    public OutputStream getOutputStream(String storageId,
-                                        String repositoryId,
-                                        String path)
-        throws IOException
+    protected boolean isChecksum(String path)
     {
-        Storage storage = getConfiguration().getStorage(storageId);
-        Repository repository = storage.getRepository(repositoryId);
-        
-        NugetHierarchicalArtifactCoordinates coordinates = new NugetHierarchicalArtifactCoordinates(path);
-        ArtifactFile artifactFile = new ArtifactFile(repository, coordinates);
-        artifactFile.createParents();
-
-        return new ArtifactFileOutputStream(artifactFile);
+        return path.endsWith("nupkg.sha512");
```

Comment:
Actually, we have a class called `ArtifactFileUtils`. I think the `isChecksum` method in it should simply be changed to support and {{SHA}} algorithm via a regex. We could then use that method instead. :)